test/amperity/gocd/secret/vault/plugin_api.clj

```diff
@@ -79,3 +79,42 @@
           (:content_type body))
       (is (:data body))
       (is (= status 200)))))
+
+
+(deftest secrets-lookup
+  (testing "Can look up secrets stored in vault given a well formed request"
+    (let [result (plugin/handle-request
+                   (mock-client)
+                   "go.cd.secrets.secrets-lookup"
+                   {:configuration {}
+                    ;; The keys will likely be string in the http vault client instance,
+                    ;; but this is easier for testing.
+                    :keys          [:batman :hulk :wonder-woman]})
+          body (:response-body result)
+          status (:response-code result)
+          _ (:response-headers result)]
+      (is (= [{:key :batman :value "Bruce Wayne"}
+              {:key :hulk :value "Bruce Banner"}
+              {:key :wonder-woman :value "Diana Prince"}]
+             body))
+      (is (= status 200))))
+  (testing "Fails cleanly when looking up secrets that don't exist"
+    (let [result (plugin/handle-request (mock-client) "go.cd.secrets.secrets-lookup"
+                                        {:configuration {}
+                                         :keys          [:dr-who :jack-the-ripper]})
+          body (:response-body result)
+          status (:response-code result)
+          _ (:response-headers result)]
+      (is (= {:message "Unable to resolve key :dr-who"}
+             body))
+      (is (= status 404))))
+  (testing "Fails cleanly when other lookup error occurs"
+    (let [result (with-redefs [vault.core/read-secret (fn [& _] (throw (ex-info "Mock Exception" {})))]
```

Comment:
You don't need `with-redefs` here, since the vault client methods are all protocols. Instead, you should `(reify vault.core/SecretClient ,,,)` with the `read-secret` behavior you want.